### DISCUSSION TITLE
How to get recursive data?

### DISCUSSION THREAD

**Discussion by SyedMSawaid** [pocketbase/pocketbase#1134] [Q&A]
Suppose I have a model with the following schema.
```
class Task {
       id: string;
       subtasks: Task[];
}
```

Now, every task can have multiple subtasks. This creates a tree. Is there any way I can get the whole tree in a single call?
Or do I have to call it individually?

**Comment by ganigeorgiev** [comment]
If `subtask` is a `relation` field, you can use the `?expand` query parameter.
More info and examples could be found in https://pocketbase.io/docs/expanding-relations/.

**Reply by dschissler** [reply]
It seems that I'm going to have to go back through the entire documentation to see how it's improved.  I'm kind of lazy and I'd like the patterns just handed to me instead of having to invent it.

**Comment by ProfEibe** [comment]
I have an additional question to this topic.

Is it possible to recursivly call subtasks of subtasks of subtasks and so on with one expand parameter?
or do I have to do expand: subtask(task).subtask(task).subtask(task)....

**Reply by ganigeorgiev** [reply]
I'm not entirely sure that I understand your question but if you need to recursively expand relations you'll have to specify explicitly the nested relations.

If you need further help, please create a new discussion with more details about your collections structure.

**Comment by Lex-2008** [comment]
You can probably create a custom _View_ with a _Recursive Common Table Expression_ (Recursive CTE). It's explained here: https://www.sqlite.org/lang_with.html

But, depending on size of your data, I believe it might be easier to have multiple individual queries.
